pkg/filter.go

```diff
@@ -0,0 +1,103 @@
+//
+// DISCLAIMER
+//
+// Copyright 2020 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Gergely Brautigam
+//
+
+package pkg
+
+import "github.com/arangodb/go-driver"
+
+// filterDatabases takes a list of databases and filters it according to the set up
+// included and excluded filters.
+func (c *copier) filterDatabases(dbs []driver.Database) []driver.Database {
+	names := make([]string, 0)
```

Comment:
Filter list of databases, not create list of names, then filter names, then rebuild database list.
Use isIncluded (see below)